src/Compilers/Core/MSBuildTask/CanonicalError.cs

```diff
@@ -72,76 +72,69 @@ internal static class CanonicalError
-        private static readonly Regex s_lineColColFromLocation = new Regex  // Example: line,col-col
-            (
-                "^"                                              // Beginning of line
-                + "(?<LINE>[0-9]*)"                               // Match any number.
-                + ","                                             // Comma
-                + "(?<COLUMN>[0-9]*)"                             // Match any number.
-                + "-"                                             // Dash
-                + "(?<ENDCOLUMN>[0-9]*)"                          // Match any number.
-                + "$",                                            // End-of-line
-                RegexOptions.IgnoreCase
-            );
+        // Example: line,col-col
+        private static readonly Regex s_lineColColFromLocation = new Regex(@"
+                ^                                           # Beginning of line
+                (?<LINE>[0-9]*)                             # Match any number.
+                ,                                           # Comma
+                (?<COLUMN>[0-9]*)                           # Match any number.
+                -                                           # Dash
+                (?<ENDCOLUMN>[0-9]*)                        # Match any number.
+                $                                           # End-of-line",
+                RegexOptions.IgnoreCase | RegexOptions.IgnorePatternWhitespace);
 
         // Matches location that is line,col,line,col
-        private static readonly Regex s_lineColLineColFromLocation = new Regex      // Example: line,col,line,col
-            (
-                "^"                                              // Beginning of line
-                + "(?<LINE>[0-9]*)"                               // Match any number.
-                + ","                                             // Comma
-                + "(?<COLUMN>[0-9]*)"                             // Match any number.
-                + ","                                             // Dash
-                + "(?<ENDLINE>[0-9]*)"                            // Match any number.
-                + ","                                             // Dash
-                + "(?<ENDCOLUMN>[0-9]*)"                          // Match any number.
-                + "$",                                            // End-of-line
-                RegexOptions.IgnoreCase
-            );
+        // Example: line,col,line,col
+        private static readonly Regex s_lineColLineColFromLocation = new Regex(@"
+                ^                                           # Beginning of line
+                (?<LINE>[0-9]*)                             # Match any number.
+                ,                                           # Comma
+                (?<COLUMN>[0-9]*)                           # Match any number.
+                ,                                           # Dash
```

Comment:
`#Comma?`

Reply:
just preserving the original comments :)   but i agree that should probably change.

Reply:
Fixed.